packages/eslint-config-wpcalypso/react/index.js

```diff
@@ -9,22 +9,23 @@ module.exports = {
 	},
 	plugins: [ 'react', 'react-hooks', 'wpcalypso' ],
 	rules: {
-		'react/jsx-curly-spacing': [ 2, 'always' ],
-		'react/jsx-no-duplicate-props': 2,
-		'react/jsx-no-target-blank': 2,
-		'react/jsx-no-undef': 2,
-		'react/jsx-tag-spacing': 2,
-		'react/jsx-uses-react': 2,
-		'react/jsx-uses-vars': 2,
-		'react/no-danger': 2,
-		'react/no-deprecated': 2,
-		'react/no-did-mount-set-state': 2,
-		'react/no-did-update-set-state': 2,
-		'react/no-is-mounted': 2,
-		'react/no-string-refs': 2,
-		'react/prefer-es6-class': 2,
-		'react/react-in-jsx-scope': 2,
-		'react-hooks/rules-of-hooks': 2,
-		'react-hooks/exhaustive-deps': 1,
+		'react/jsx-boolean-value': [ 'error', 'never' ],
+		'react/jsx-curly-spacing': [ 'error', 'always' ],
+		'react/jsx-no-duplicate-props': 'error',
+		'react/jsx-no-target-blank': 'error',
+		'react/jsx-no-undef': 'error',
+		'react/jsx-tag-spacing': 'error',
+		'react/jsx-uses-react': 'error',
+		'react/jsx-uses-vars': 'error',
```

Comment:
`jsx-uses-vars` and `jsx-no-undef` comes from ancient times (2015) when [ESLint refused to support JSX natively](https://eslint.org/blog/2015/03/eslint-0.17.0-released#changes-to-jsxreact-handling). And, e.g., a `<Hello>` tag was not detected as usage of the `Hello` variable (marking it as unused), or detected as usage of an undefined `Hello` variable.

What is the situation today? Can the vanilla `no-undef` and `no-unused-vars` rules detect the same thing? My little testing shows that `no-unused-vars` finds an usage inside JSX, but `no-undef` doesn't work with JSX: I can use a `<Blah />` tag without ever defining `Blah`.

Reply:
What's the conclusion? That `react/jsx-uses-vars` is redundant and can be removed but `jsx-no-undef` cannot?

Reply:
As @scinos found out, both `jsx-uses-vars` and `jsx-no-undef` are still needed.

`jsx-uses-vars` doesn't report any errors on its own, but works together with `no-unused-vars`: it marks JSX usages as used using ESLint API, so that `no-unused-vars` doesn't report them.

`jsx-no-undef` does its own check and reporting, independently from `no-undef`.